### PR TITLE
fix(openclaw): resolve base58 cryptoId recipients via the directory

### PR DESCRIPTION
**Pull request by oxoxDev**
## Summary

`message send <recipient>` failed for a base58 cryptoId/agentId recipient — `isPublicKey` matched any 42–46 char `[A-Za-z0-9+/]` string, so a cryptoId (also ~44 chars, no `+`/`/`/`=`) was misclassified as a raw base64 encryption key. The send then fetched `/keys/<cryptoId>/bundle` directly and 404'd, instead of resolving the recipient's directory card to its advertised encryption key.

## Fix

- `isPublicKey` now requires the trailing base64 `=` pad (a 32-byte Ed25519 key is always 44 chars ending in `=`); a base58 cryptoId never has it.
- A bare cryptoId/agentId is resolved via `directory.getAgent` → the card's `encryptionPublicKey` (falling back to the card's `publicKey`).
- `message send` now accepts all three recipient forms: raw base64 key, `@handle`, and cryptoId.

## Tests

Added `messaging.test.ts` (node:test): `isPublicKey` classification (base64 key / cryptoId / @handle) and `resolveRecipientKey` across all three paths incl. the encryption-key fallback. Full plugin suite: 57 passing.

## Verification

Reproduced + fixed live against staging: sending to a cryptoId previously 404'd on the bundle fetch; after the fix it resolves the card and delivers an E2E message (PREKEY_BUNDLE → CIPHERTEXT).

## Summary by CodeRabbit

* **New Features**
  * Added public utilities to validate raw base64 encryption keys and to resolve recipients from `@handles`, agent/crypto IDs, and key formats.
* **Refactor**
  * Improved recipient/key resolution flow with clear handling for `@handle` vs bare IDs and enhanced fallback selection of the best available encryption public key.
* **Tests**
  * Added Node test coverage for key validation and recipient resolution, including edge cases and metadata fallback behavior.

### CODE REVIEW

File: sdk/plugin-openclaw/src/messaging.test.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import assert from "node:assert/strict";
+import test from "node:test";
+
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply sdk/plugin-openclaw/src/messaging.test.ts:3]: Keeping `node:test` here. `sdk/plugin-openclaw` runs `tsc && node --test dist/*.test.js`, has no `vitest` dependency or config, and all 5 existing test files in this package (wallet/moonpay/shared/group-messaging/config) use `node:test`. `messaging.test.ts` follows that established convention so it runs under the package's `pnpm test`. Converting just this file would break it (no `vitest` dep → tsc fails); migrating the whole package to Vitest (+ config) is a separate change, out of scope for this recipient-resolution fix. Happy to open a follow-up to migrate the package if maintainers want it.

File: sdk/plugin-openclaw/src/messaging.ts [hunk 2]
```diff
@@ -43,39 +43,78 @@ const ENCRYPTION_PUBLIC_KEY_METADATA = "encryptionPublicKey";
...
+      resolved.agent?.metadata?.[ENCRYPTION_PUBLIC_KEY_METADATA],
+      resolved.agent?.publicKey,
+      resolved.identity?.publicKey,
+      recipient,
+    );
   }
-  return address;
+
+  // A bare base58 cryptoId / agentId: look up its directory card directly.
+  const card = await client.directory.getAgent(recipient);
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply sdk/plugin-openclaw/src/messaging.ts]: Good catch — fixed in 4416db2. Only a bare base58 cryptoId/agentId (32-44 base58 chars) now takes the direct `getAgent` lookup; any other bare value is treated as a handle, normalized to `@name`, and resolved via `directory.resolve` — so `message send iris …` works again. Added a regression test covering the bare-handle path.
